src/main/java/com/icthh/xm/uaa/security/ldap/UaaLdapUserDetailsContextMapper.java

```diff
@@ -67,29 +70,29 @@ private void createUser(DirContextOperations ctx,
         userLogin.setLogin(username);
         userLogin.setTypeKey(UserLoginType.NICKNAME.getValue());
         userDTO.setLogins(Collections.singletonList(userLogin));
-        userDTO.setRoleKey(mapRole(roleConf, authorities).orElse(roleConf.getDefaultRole()));
+        userDTO.setAuthorities(mapRole(roleConf, authorities).orElse(List.of(roleConf.getDefaultRole())));
         userService.createUser(userDTO);
     }
 
     private void updateUser(DirContextOperations ctx, User user, Collection<? extends GrantedAuthority> authorities) {
         TenantProperties.Ldap.Role roleConf = ldapConf.getRole();
-        String mappedRole = mapRole(roleConf, authorities)
-            .orElseGet(() -> defaultIfEmpty(user.getRoleKey(), roleConf.getDefaultRole()));
-        log.info("Mapped role from ldap [{}], current role [{}]", mappedRole, user.getRoleKey());
+        List<String> mappedRoles = mapRole(roleConf, authorities)
+            .orElseGet(() -> isEmpty(user.getAuthorities()) ? List.of(roleConf.getDefaultRole()) : user.getAuthorities());
+        log.info("Mapped role from ldap [{}], current role [{}]", mappedRoles, user.getAuthorities());
         String imageUrl = parseImageUrl(ctx);
 
-        if (needUpdate(mappedRole, user, imageUrl)) {
+        if (needUpdate(mappedRoles, user, imageUrl)) {
             user.setImageUrl(imageUrl);
-            user.setRoleKey(mappedRole);
+            user.setAuthorities(mappedRoles);
             userService.saveUser(user);
         }
     }
 
-    private boolean needUpdate(String mappedRole, User user, String imageUrl) {
-        return !mappedRole.equals(user.getRoleKey()) || !StringUtils.equals(user.getImageUrl(), imageUrl);
+    private boolean needUpdate(List<String> mappedRoles, User user, String imageUrl) {
+        return !isEqualCollection(mappedRoles, user.getAuthorities()) || !StringUtils.equals(user.getImageUrl(), imageUrl);
     }
 
-    private Optional<String> mapRole(TenantProperties.Ldap.Role roleConf, Collection<? extends GrantedAuthority> authorities) {
+    private Optional<List<String>> mapRole(TenantProperties.Ldap.Role roleConf, Collection<? extends GrantedAuthority> authorities) {
```

Comment:
List is already container. do not need Optional here. Use just a List<String>

Reply:
fixed